modern-publish-plugin.md

```diff
@@ -0,0 +1,211 @@
+# Modern Gradle Publish Plugin
+
+
+## Goals
+
+- less magic configuration, more explicit APIs
+- provide APIs for everything represented through properties
+- remove or deprecate APIs that are already provided by Gradle
+- continue providing a way to get started with minimal configuration
+
+## New Structure
+
+For compatibility purposes the current `com.vanniktech.maven.publish` plugin
+behavior is kept as it is, with the exception of APIs that we decide to remove. 
+Most of the goals are solved through a set of new plugins for specific 
+platforms. These plugins provide APIs that replace the magic setup mechanisms
+of the current plugin. To avoid having to maintain two implentations the old
+plugin will internally just apply the new plugins and use their public APIs. 
+Over time we can decide if `com.vanniktech.maven.publish` should be deprecated
+or removed.
+
+### Plugins
+
+Specific plugins for each supported project type:
+- `com.vanniktech.maven.publish.java`
+- `com.vanniktech.maven.publish.gradle`
+- `com.vanniktech.maven.publish.android`
+- `com.vanniktech.maven.publish.kotlin.js`
+- `com.vanniktech.maven.publish.kotlin.mpp`
+
+There are two reasons for going with individual plugins. 
```

Comment:
What are the two reasons?

Reply:
- In general it allows us to separate the code better
- The main reason is that even with using `plugins.withId` and `afterEvaluate` there will be weird cases when plugins are defined in a certain order. For example let's say you define the publish plugin and then the android plugin and you have a `plugins.withId("...publish")` to call the `androidPublishCompat` API. The publish plugin would add the API only after it detects the Android plugin but that happens after the user's `plugins.withId`.